src/bp/core/services/dialog/flow/service.ts

```diff
@@ -26,9 +30,23 @@ interface FlowModification {
   payload?: any
 }
 
+interface FlowMutex {
+  lastModifiedBy: string
+  lastModifiedAt: Date
+  remainingSeconds?: number // backend calculate this because all clients time might be wrong
```

Comment:
This could be a getter instead?  Would prevent you having a `_setRemainingSeconds` logic etc

Reply:
Well I need this object to be serialized and sent to the frontend so clients know the remaining seconds before unlocking the flow.